File: 705. Design HashSet/Solution.py
class MyHashSet:

    def __init__(self):
        self.buckets = [[] for i in range(31)]

    def add(self, key: int) -> None:
        hash = key % 31
        flag = False
        first = -1
        for i, x in enumerate(self.buckets[hash]):
            if x == key:
                flag = True
            if x == -1 and first == -1:
                first = i
        if not flag:
            if first == -1:
                self.buckets[hash].append(key)
            else:
                self.buckets[hash][first] = key

    def remove(self, key: int) -> None:
        hash = key % 31
        for i, x in enumerate(self.buckets[hash]):
            if x == key: self.buckets[hash][i] = -1

    def contains(self, key: int) -> bool:
        hash = key % 31
        for x in self.buckets[hash]:
            if x == key: return True
        return False


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)

File: 705. Design HashSet/Solution.cpp
class MyHashSet {
public:
    vector<vector<int>> buckets;
    MyHashSet() {
        for (int i = 0; i < 31; i++) {
            vector<int> bucket;
            buckets.push_back(bucket);
        }
    }
    
    void add(int key) {
        int hash = key % 31, first = -1, n = buckets[hash].size();
        bool flag = false;
        for (int i = 0; i < n; i++) {
            if (buckets[hash][i] == key) flag = true;
            if (buckets[hash][i] == -1) first = i;
        }
        if (!flag) {
            if (first == -1) {
                buckets[hash].push_back(key);
            } else {
                buckets[hash][first] = key;
            }
        }
    }
    
    void remove(int key) {
        int hash = key % 31, n = buckets[hash].size();
        for (int i = 0; i < n; i++) {
            if (buckets[hash][i] == key) buckets[hash][i] = -1;
        }
    }
    
    bool contains(int key) {
        int hash = key % 31, n = buckets[hash].size();
        for (int i = 0; i < n; i++) {
            if (buckets[hash][i] == key) return true;
        }
        return false;
    }
};

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet* obj = new MyHashSet();
 * obj->add(key);
 * obj->remove(key);
 * bool param_3 = obj->contains(key);
 */
